rotkehlchen/data/import_utils.py

```diff
@@ -0,0 +1,283 @@
+                        to_asset=A_USD,
+                        timestamp=timestamp,
+                    )
+                except NoPriceForGivenTimestamp:
+                    log.warning(f'Couldn\'t find price of {row["Coin"]} on {timestamp}')
+                    return []
+                row['usd_value'] = row['Change'] * price
+
+        def ops_key(x: BinanceCsvRow) -> bool:
+            return x['Operation'] == 'Fee'
+        rows_by_operation = {}
+        for is_fee, rows in groupby(sorted(data, key=ops_key), key=ops_key):
+            rows_by_operation[is_fee] = sorted(
+                rows, key=lambda x: x['Change'] if same_assets else x['usd_value'],
+            )
+
+        trade_rows = []
+        while len(rows_by_operation[False]) > 0:
+            cur_batch = [rows_by_operation[False].pop(0), rows_by_operation[False].pop(-1)]
+            if True in rows_by_operation:
+                cur_batch.append(rows_by_operation[True].pop(0))
+            trade_rows.append(cur_batch)
+
+        prepared_trade_dicts = []
+        for t_rows in trade_rows:
+            cur_data: BinanceCsvRow = defaultdict(lambda: None)
+            for row in t_rows:
+                cur_asset = row['Coin']
+                amount = row['Change']
+                if row['Operation'] == 'Fee':
+                    cur_data['fee_asset'] = cur_asset
+                    cur_data['fee_amount'] = Fee(amount)
+                else:
+                    cur_data['trade_type'] = TradeType.SELL if row['Operation'] == 'Sell' else TradeType.BUY  # noqa:  E501
+                    if amount < 0:
+                        cur_data['from_asset'] = cur_asset
+                        cur_data['from_amount'] = AssetAmount(-amount)
+                    else:
+                        cur_data['to_asset'] = cur_asset
+                        cur_data['to_amount'] = amount
+            prepared_trade_dicts.append(cur_data)
+
+        def trades_key(t_dict: Dict) -> str:
+            return ';'.join(sorted([f'{x[0]}: {x[1]}' for x in t_dict.items()]))
+
+        grouped_trades_dicts = []
+        for _, grouped_trades in groupby(sorted(prepared_trade_dicts, key=trades_key), key=trades_key):  # noqa: E501
+            grouped_trades_dicts.append(list(grouped_trades))
+
+        unique_trade_dicts = []
+        for trades_dict_group in grouped_trades_dicts:
+            result_trade_dict = trades_dict_group[0]
+            for trade_dict in trades_dict_group[1:]:
+                result_trade_dict['to_amount'] += trade_dict['to_amount']
+                result_trade_dict['from_amount'] += trade_dict['from_amount']
+                if result_trade_dict['fee_amount'] is not None:
+                    result_trade_dict['fee_amount'] += trade_dict['fee_amount']
+            unique_trade_dicts.append(result_trade_dict)
+
+        trades = []
+        for trade_dict in unique_trade_dicts:
+            rate = FVal(trade_dict['to_amount']) / FVal(trade_dict['from_amount'])
```

Comment:
What happens if for some reason the data is not correct and you have here a division by zero error. Do you handle it in other place? I believe you would want to skip the row and notify the user

Reply:
Yeah, that's a good point, thanks! Added validation